plugins/volumeFilter_test.go

```diff
@@ -2,224 +2,293 @@ package plugins
-	accountConfigVolumeQuery, e := makeDailyVolumeByDateQuery(db, "", "native", "native", accountIDs, []string{})
-	if e != nil {
-		t.Log("could not make account config volume query")
-		return
-	}
-
-	accountMarketConfigVolumeQuery, e := makeDailyVolumeByDateQuery(db, "", "native", "native", accountIDs, marketIDs)
-	if e != nil {
-		t.Log("could not make account and market config volume query")
-		return
-	}
+	testAssetDisplayFn := model.MakeSdexMappedAssetDisplayFn(map[model.Asset]hProtocol.Asset{model.Asset("XLM"): testNativeAsset})
 
 	testCases := []struct {
 		name           string
-		configValue    string
-		exchangeName   string
-		tradingPair    *model.TradingPair
 		assetDisplayFn model.AssetDisplayFn
-		baseAsset      hProtocol.Asset
-		quoteAsset     hProtocol.Asset
-		db             *sql.DB
 		config         *VolumeFilterConfig
 
 		wantError        error
 		wantSubmitFilter SubmitFilter
 	}{
-		// Note: config can be additional market IDs, make sure to test additionalMarketIds and optionalAccountIDs
+		// TODO DS Confirm the empty config fails once validation is added to the constructor
 		{
-			name:             "failure - display base",
-			configValue:      "",
-			exchangeName:     "",
-			tradingPair:      &model.TradingPair{Base: "FAIL", Quote: ""},
-			assetDisplayFn:   testAssetDisplayFn,
-			baseAsset:        hProtocol.Asset{},
-			quoteAsset:       hProtocol.Asset{},
-			db:               db,
-			config:           &VolumeFilterConfig{},
-			wantError:        fmt.Errorf("could not convert base asset (FAIL) from trading pair via the passed in assetDisplayFn: cannot recognize the asset FAIL"),
-			wantSubmitFilter: nil,
+			name:      "empty config",
+			config:    &VolumeFilterConfig{},
+			wantError: nil,
+			wantSubmitFilter: &volumeFilter{
+				name:                   "volumeFilter",
+				baseAsset:              testNativeAsset,
+				quoteAsset:             testNativeAsset,
+				config:                 &VolumeFilterConfig{},
+				dailyVolumeByDateQuery: mustMakeTestDailyVolumeQuery([]string{}, []string{}),
+			},
 		},
 		{
-			name:             "failure - display quote",
-			configValue:      "",
-			exchangeName:     "",
-			tradingPair:      &model.TradingPair{Base: "XLM", Quote: "FAIL"},
-			assetDisplayFn:   testAssetDisplayFn,
-			baseAsset:        testNativeAsset,
-			quoteAsset:       hProtocol.Asset{},
-			config:           &VolumeFilterConfig{},
-			wantError:        fmt.Errorf("could not convert quote asset (FAIL) from trading pair via the passed in assetDisplayFn: cannot recognize the asset FAIL"),
-			wantSubmitFilter: nil,
+			name: "non nil cap in base, nil cap in quote",
+			config: &VolumeFilterConfig{
+				SellBaseAssetCapInBaseUnits: pointy.Float64(1.0),
+				mode:                        volumeFilterModeExact,
+				additionalMarketIDs:         []string{},
+				optionalAccountIDs:          []string{},
+			},
+			wantError: nil,
+			wantSubmitFilter: &volumeFilter{
+				name:       "volumeFilter",
```

Comment:
name and base/quote assets are not changing across the test cases -- can we move these to be inside the test?

we also don't need to re-specify the config inside the `wantSubmitFilter` since that's exactly the same as what we pass in -- in the test we can check that `assert.Equal(t, k.config, submitFilter.config)`
in fact I think the test case should only take in 1 want value: `wantDailyVolumeByDateQuery`.
we don't need wantError either because none of our cases will produce an error

Take a look at `TestFilterOrdersByVolume` for an example of a test where the testCases does not match the function inputs and outputs exactly to make them more readable and reduce boilerplate ([link](https://github.com/stellar/kelp/blob/master/plugins/mirrorStrategy_test.go#L107))

Reply:
additionally -- we can have two test runs for each case specified here inside the `t.Run()` function below -- one with nil base cap, and the other with a nil quote cap. that way we don't need to specify 2x cases here.

we can do the same for exact and ignore -- since those aren't modified either. so this will be a nested for loop for each case specified here.

something like this:
```
for test cases:
    for _, m := range []volumeFilterMode{exact, ignore} {
        baseCapConfig := &VolumeFilterConfig{
            SellBaseAssetCapInBaseUnits: point.Float64(1.0),
            SellBaseAssetCapInQuoteUnits: nil,
            mode: m,
            additionalMarketIDs: k.additionalMarketIDs,
            optionalAccountIDs: k.optionalAccountIDs,
        }

        quoteCapConfig := &VolumeFilterConfig{
            SellBaseAssetCapInBaseUnits: nil,
            SellBaseAssetCapInQuoteUnits: point.Float64(1.0),
            mode: m,
            additionalMarketIDs: k.additionalMarketIDs,
            optionalAccountIDs: k.optionalAccountIDs,
        }

        for _, c := range []*VolumeFilterConfig{baseCapConfig, quoteCapConfig} {
            isBaseCap := c.SellBaseAssetCapInBaseUnits != nil
            name := fmt.Sprintf("%s_%s_%v", k.namePrefix, m.String(), isBaseCap)
            t.Run(name, func(t *testing.T) {
                // refer to TestFilterOrdersByVolume for how to convert test case inputs to function inputs and function outputs to test case outputs
            })
        }
    }
```

does this make sense?

Reply:
this will reduce the test cases significantly

for the empty config, we can have a special test since it doesn't conform to this model